packages/commercetools/theme/tests/e2e/integration/e2e-checkout-billing-validation.spec.ts

```diff
@@ -0,0 +1,123 @@
+import requests, { CreateCartResponse } from '../api/requests';
+import page from '../pages/factory';
+import generator from '../utils/data-generator';
+
+context(['regression'], 'Checkout - Billing', () => {
+  beforeEach(function () {
+    cy.fixture('test-data/e2e-checkout-billing-validation').then((fixture) => {
+      this.fixtures = {
+        data: fixture
+      };
+    });
+  });
+
+  it('Should successfully save address - guest customer', function () {
+    const data = this.fixtures.data[this.test.title];
+    requests.createCart().then((response: CreateCartResponse) => {
+      requests.addToCart(response.body.data.cart.id, data.product);
+      requests.updateCart(response.body.data.cart.id, { addresses: { shipping: data.customer.address.shipping }});
+    });
+    page.checkout.shipping.visit();
+    page.checkout.shipping.selectShippingButton.click();
+    page.checkout.shipping.shippingMethods.contains(data.shippingMethod).click();
+    page.checkout.shipping.continueToBillingButton.click();
+    page.checkout.billing.heading.should('be.visible');
+    page.checkout.billing.fillForm(data.customer);
+    page.checkout.billing.continueToPaymentButton.click();
+    page.checkout.payment.heading.should('be.visible');
```

Comment:
To have less code, you can destructure `page.checkout` object like this:

Look at example below
<img width="778" alt="Zrzut ekranu 2021-07-2 o 15 43 14" src="https://user-images.githubusercontent.com/24594740/124283468-38b02f00-db4c-11eb-8554-1a2d998cc3ed.png">